src/main/java/com/example/petbuddybackend/service/offer/OfferService.java

```diff
@@ -0,0 +1,278 @@
+
+        if(animalAttributesLists.stream().anyMatch(animalAttributes -> animalAttributes.equals(newAnimalAttributes))) {
+            throw new OfferConfigurationAlreadyExistsException(MessageFormat.format(
+                            "Offer configuration with animal attributes {0} already exists",
+                            newAnimalAttributes.stream()
+                                    .map(AnimalAttribute::toString)
+                                    .collect(Collectors.joining(", "))
+                    ));
+        }
+
+    }
+
+    private OfferConfiguration createConfiguration(OfferConfigurationDTO offerConfiguration, Offer offer) {
+        OfferConfiguration newOfferConfiguration = OfferConfiguration.builder()
+                .dailyPrice(offerConfiguration.dailyPrice())
+                .description(offerConfiguration.description())
+                .offer(offer)
+                .build();
+
+        List<OfferOption> offerOptions =
+                createOfferOptionsForConfiguration(offerConfiguration.selectedOptions(), newOfferConfiguration);
+
+        newOfferConfiguration.setOfferOptions(offerOptions);
+        return newOfferConfiguration;
+    }
+
+    private List<OfferOption> createOfferOptionsForConfiguration(Map<String, List<String>> selectedOptions,
+                                                                 OfferConfiguration offerConfiguration) {
+
+        List<OfferOption> offerOptions = new ArrayList<>();
+        for(Map.Entry<String, List<String>> entry : selectedOptions.entrySet()) {
+            String attributeName = entry.getKey();
+            for(String attributeValue : entry.getValue()) {
+                AnimalAttribute animalAttribute = getAnimalAttribute(offerConfiguration.getOffer().getAnimal().getAnimalType(),
+                        attributeName, attributeValue);
+                offerOptions.add(createOfferOption(animalAttribute, offerConfiguration));
+            }
+        }
+        return offerOptions;
+
+    }
+
+    private OfferOption createOfferOption(AnimalAttribute animalAttribute, OfferConfiguration offerConfiguration) {
+        return OfferOption.builder()
+                .animalAttribute(animalAttribute)
+                .offerConfiguration(offerConfiguration)
+                .build();
+    }
+
+    private AnimalAttribute getAnimalAttribute(String animalType, String attributeName, String attributeValue) {
+        return animalAttributeRepository.findByAnimal_AnimalTypeAndAttributeNameAndAttributeValue(animalType, attributeName, attributeValue)
+                .orElseThrow(() -> new NotFoundException("Animal attribute with name " + attributeName + " and value " + attributeValue + " not found"));
+    }
+
+    private Caretaker getCaretaker(String caretakerEmail) {
+        return caretakerRepository.findById(caretakerEmail)
+                .orElseThrow(() -> new NotFoundException("Caretaker with email " + caretakerEmail + " not found"));
+    }
```

Comment:
Przeniósłbym do CaretakerService (o ile już nie ma takiej metody)

Reply:
zdecydowanie

Reply:
a co myslisz nad rozdzieleniem OfferService na OfferService i OfferConfigurationService, wtedy OfferService korzystalby z OfferConfigurationService. Czy przesada i moze byc jak jest?

Reply:
Wydaje mi się że one są na tyle powiązane że mogą być razem. Ewentualnie jakbyś chciał coś rozdzielać to może animal, animal amenity itp. ale to jak uważasz